### ISSUE TITLE
brew complains about using pre-release version of xcode 8.1 on 10.12.1

### ISSUE THREAD

**Issue by sunilshah** [Homebrew/homebrew-core#6359]
If Homebrew was updated on Aug 10-11th 2016 and `brew update` always says `Already up-to-date.` you need to run: `cd "$(brew --repo)" && git fetch && git reset --hard origin/master && brew update`.

# Please follow the general troubleshooting steps first:

- [x] Ran `brew update` and retried your prior step?
- [x] Ran `brew doctor`, fixed as many issues as possible and retried your prior step?
- [ ] If you're seeing permission errors tried running `sudo chown -R $(whoami) $(brew --prefix)`?

_You can erase any parts of this template not applicable to your Issue._

### Bug reports:

Please replace this line with a brief summary of your issue **AND** if reporting a build issue include the link from:

`brew gist-logs <formula>`
(where `<formula>` is the name of the formula that failed to build).

### Formula Requests:

**Please note by far the quickest way to get a new formula into Homebrew is to file a [Pull Request](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md).**

We will consider your request but it may be closed if it's something we're not actively planning to work on.

**Comment by sunilshah** [comment]
I used Apple Developer site download link to update Xcode from 8.0 to 8.1 on 10.12.1.  I did this to get faster download on one machine.  

On the other machine I used the App store to download.  Brew on that machine does not complain about a pre-release version, only for the machine that used download link. 

The versions of Xcode are release 8.1 (build 8B62) on both yet brew complains that it is a pre-release version for the first machine.

xcode-select --version  and gcc -v outputs are also the same on the two machines. How does brew find out whether the version of Xcode is pre-release?
